### ISSUE TITLE
NetCoreAppSettings doesn't support nested keys consistently

### ISSUE THREAD

**Issue by logiclabs** [ServiceStack/Issues#657]
The [NetCoreAppSettings](https://github.com/ServiceStack/ServiceStack/blob/master/src/ServiceStack/NetCoreAppSettings.cs) has some inconsistency with using a nested key in the app settings i.e.
```javascript
{
 "A": { 
          "A1": "A_A1_Value", 
          "A2": { 
                "A": "A_A2_A_Value" 
           } 
       },
 "B": "B_Value"
}
```

- `GetString(key)` and `Get(key)` work correctly for all key values (so can return A:A2:A).
- `GetAll()`,  `GetAllKeys()`, `GetList(key)`, `GetDictionary(key)` and `Exists(key)` can only see top-level keys (so in above "B") so only return the top level keys or fail.
- `Get<T>(key,  default)` fails to find a nested subkey so always returns default.

The existing unit tests for NetCoreAppSettings only cover a top-level keys, so this isn't caught

The issue seems to be some of the methods above using `Configuration.GetChildren()` and just scanning for matching keys, but `GetChildren()` only returns top the level items.

I'm using v.5.4.1 from MyGet.  

## Tests:
https://gist.github.com/logiclabs/9d38c3b74f4cd2c649a6c1c3676c3f67

## Failing Test Cases: 
![image](https://user-images.githubusercontent.com/1760571/54072598-cdecd280-4274-11e9-9616-de64e30a3e0f.png)

**Comment by logiclabs** [comment]
One solution that might work is changing `Configuration.GetChildren().FirstOrDefault(x => x.Key == name)` usage to `Configuration.GetSection(key)`.  Null checks would need to be switched to check if the empty sections Value is null.

**Comment by mythz** [comment]
I've resolved most of these tests with [this commit](https://github.com/ServiceStack/ServiceStack/commit/569484d09e0b03e85a9f20f3e21ed188247a380b).

Some of these tests broke compatibility with the other AppSetting providers, e.g. GetAllKeys is only going to return the top-level keys, if you're after different behavior to return all sub keys you'll need to [access the `IConfiguration` object directly](https://github.com/ServiceStack/ServiceStack/blob/569484d09e0b03e85a9f20f3e21ed188247a380b/tests/ServiceStack.Common.Tests/Configuration/NetCoreAppSettingsTests.cs#L94), e.g:

```csharp
var appSettings = (GetAppSettings() as NetCoreAppSettings).Configuration;
var allKeyValues = appSettings.AsEnumerable();
foreach (var key in Settings.Keys)
{
    Assert.That(allKeyValues.Any(x => x.Key == key));
}
```

Also you wont be able to get the exact keys in your MemoryConfigurationSource via .NET Core's `IConfiguration` abstraction as it will return additional implied keys so the test can only check for existence of the existing keys.

This change is available from the latest v5.4.1 on MyGet as you've already got v5.4.1 installed you'll need to [clear your NuGet packages cache](https://docs.servicestack.net/myget#redownloading-myget-packages).

thx for the tests!

**Comment by logiclabs** [comment]
Wow, so quick! Thanks.. I thought changing GetAllKeys behaviour might be an issue, but personally don't have any need for it (yet!).
